src/Language/PureScript/AST/Declarations.hs

```diff
@@ -188,9 +194,15 @@ data Declaration
   --
   | ExternDataDeclaration (ProperName 'TypeName) Kind
   -- |
-  -- A fixity declaration (fixity data, operator name, value the operator is an alias for)
+  -- A fixity declaration
+  -- (fixity data, operator name, value the operator is an alias for)
   --
-  | FixityDeclaration Fixity String (Maybe (Qualified FixityAlias))
+  | FixityDeclaration Fixity (OpName 'ValueOpName) (Qualified (Either Ident (ProperName 'ConstructorName)))
+  -- |
+  -- A fixity declaration for a type operator
+  -- (fixity data, operator name, type the operator is an alias for)
+  --
+  | TypeFixityDeclaration Fixity (OpName 'TypeOpName) (Qualified (ProperName 'TypeName))
```

Comment:
I'm not sure about this approach yet (separating the type fixities entirely). It might result in a ton of duplicated code, but on the other hand it will eliminate a bunch of cases where we have a multiplication of possibilities due to type/value split.

I tried some stuff out with GADTs, but it quickly spiralled into madness.